### PR TITLE
RUBY-169 - Ignore authenticator name when creating password authenticator

### PR DESCRIPTION
**Pull request by stamhankar999**
RUBY-169 - Ignore authenticator name when creating password authenticator.

NB: Not updating CHANGELOG because this is so under the hood that I don't think users would notice.

### CODE REVIEW

File: lib/cassandra/auth/providers/password.rb [hunk 1]
```diff
@@ -53,17 +53,11 @@ def initialize(username, password)
...
-        #   `org.apache.cassandra.auth.PasswordAuthenticator`
-        # @return [Cassandra::Auth::Authenticator] when `authentication_class ==
-        #    "org.apache.cassandra.auth.PasswordAuthenticator"`
-        # @return [nil] for all other values of `authentication_class`
-        def create_authenticator(authentication_class)
-          if authentication_class == PASSWORD_AUTHENTICATOR_FQCN
-            Authenticator.new(@username, @password)
-          end
+        # Returns a Password Authenticator
+        # @param authentication_class [String] ignored and deprecated
```
Comment (avalanche123) [inline lib/cassandra/auth/providers/password.rb]: it is not deprecated, just ignored, also it shouldn't default to nil

**Comment by stamhankar999** [comment]
Q: can we remove the arg to create_authenticator? I changed it to be optional for now in case a user is calling it, or in case this method signature lends itself to other authenticator impls later.

**Comment by avalanche123** [comment]
The users shouldn't be calling it, and no, we shouldn't change its signature to optional because:

* that breaks public API
* the driver always passes the authenticator name received from cassandra to this method

**Comment by stamhankar999** [comment]
Changing it to optional wouldn't break the public api; it just makes it looser so callers might or might not send that arg and things will still work; only worthwhile if over time we want to deprecate the arg. The driver could be made to *not* pass the authenticator name. So my question boils down to, "Do we want this api to eventually not take an arg, or do we want the arg because we may still use it in the future?"

**Comment by avalanche123** [comment]
It breaks public API, because now you can call `create_authenticator` without arguments. We don't want to make the argument optional as the driver always receives authenticator class from cassandra and that can always be used in a custom authenticator

**Comment by stamhankar999** [comment]
Ok, made the authentication_class arg required again.

**Comment by avalanche123** [comment]
:+1:

**Pull request merged**